### ISSUE TITLE
Typehint in DrupalExtension produces PHP Fatal error

### ISSUE THREAD

**Issue by daFish** [mglaman/phpstan-drupal#17]
Introduced in #14.

The added typehint leads to a fatal error when a parameter for a service definition is not a string (e.g. `null`, `array`).

https://github.com/mglaman/phpstan-drupal/blob/410ea5ad288455a206693ced0c890df80524c1f2/src/DependencyInjection/DrupalExtension.php#L145

I'd propose to remove the typehint until we come up with a better solution.

**Comment by mglaman** [comment]
I'd like to figure out what services cause the problem (main reason I like type hints.) Do you have examples of what causes the failure?

**Comment by mglaman** [comment]
Errr probably if there's no arguments.

**Comment by jibran** [comment]
This problem occurs for the services with the factory method.

```
Array
(
    [0] => @http_kernel.controller.argument_metadata_factory
    [1] => Array
        (
            [0] => @argument_resolver.request_attribute
            [1] => @argument_resolver.raw_parameter
            [2] => @argument_resolver.request
            [3] => @argument_resolver.psr7_request
            [4] => @argument_resolver.route_match
            [5] => @argument_resolver.default
        )

)```

**Comment by jibran** [comment]
This is the only service in `core.services.yml` which has array argument instead of string.

**Comment by mglaman** [comment]
Oh, interesting. The second argument is an array of arguments. TIL that's a thing. What I did previously was ignore these use cases (like factory) until they can be properly handled.

**Comment by mglaman** [comment]
This is now proven in #29

**Comment by mglaman** [comment]
Fixed by c7b57869cc484587e76bde2cd3542d19d9306d19, and now testable
